### PR TITLE
[SPARK-26651][SQL][DOC] Collapse notes related to java.time API

### PR DESCRIPTION
**Pull request by MaxGekk**
## What changes were proposed in this pull request?

Collapsed notes about using Java 8 API for date/timestamp manipulations and Proleptic Gregorian calendar in the SQL migration guide.

### CODE REVIEW

**Comment by MaxGekk** [comment]
@srowen @HyukjinKwon Please, have a look at the changes.

File: docs/sql-migration-guide-upgrade.md [hunk 1]
```diff
@@ -9,8 +9,6 @@ displayTitle: Spark SQL Upgrading Guide
 
 ## Upgrading From Spark SQL 2.4 to 3.0
 
-  - Since Spark 3.0, the Dataset and DataFrame API `unionAll` is not deprecated any more. It is an alias for `union`.
```
Comment (srowen) [inline docs/sql-migration-guide-upgrade.md]: Add this back?

**Review by srowen — CHANGES_REQUESTED** [review]

File: docs/sql-migration-guide-upgrade.md (hunk 1 above)
Comment (MaxGekk) [reply docs/sql-migration-guide-upgrade.md]: reverted

**Review by srowen — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**